src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs

```diff
@@ -53,6 +53,12 @@ private void InitializeContentPresenter()
 		{
 		}
 
+		/// <summary>
+		/// Determines if the current ContentPresenter is hosting a native control.
+		/// </summary>
+		/// <remarks>This is used to alter the propagation of the templated parent.</remarks>
+		internal bool IsNativeHost { get; set; }
```

Comment:
I wonder if this could be an evaluated property that looks at the _contentTemplateRoot... but this is good enough for now :)

Reply:
That's fair. The only part that I can see is that it's one less check :)